NineChronicles.Headless/Middleware/HttpMultiAccountManagementMiddleware.cs

```diff
@@ -0,0 +1,185 @@
+using Libplanet.Crypto;
+using Libplanet.Types.Tx;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using NineChronicles.Headless.Properties;
+using Serilog;
+using ILogger = Serilog.ILogger;
+
+namespace NineChronicles.Headless.Middleware
+{
+    public class HttpMultiAccountManagementMiddleware
+    {
+        private static readonly Dictionary<Address, DateTimeOffset> MultiAccountTxIntervalTracker = new();
+        private static readonly Dictionary<Address, DateTimeOffset> MultiAccountManagementList = new();
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private StandaloneContext _standaloneContext;
+        private readonly Dictionary<string, HashSet<Address>> _ipSignerList;
+        private readonly IOptions<MultiAccountManagerProperties> _options;
+
+        public HttpMultiAccountManagementMiddleware(
+            RequestDelegate next,
+            StandaloneContext standaloneContext,
+            Dictionary<string, HashSet<Address>> ipSignerList,
+            IOptions<MultiAccountManagerProperties> options)
+        {
+            _next = next;
+            _logger = Log.Logger.ForContext<HttpMultiAccountManagementMiddleware>();
+            _standaloneContext = standaloneContext;
+            _ipSignerList = ipSignerList;
+            _options = options;
+        }
+
+        private static void ManageMultiAccount(Address agent)
+        {
+            MultiAccountManagementList.Add(agent, DateTimeOffset.Now);
+        }
+
+        private static void RestoreMultiAccount(Address agent)
+        {
+            MultiAccountManagementList.Remove(agent);
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            // Prevent to harm HTTP/2 communication.
+            if (context.Request.Protocol == "HTTP/1.1")
+            {
+                context.Request.EnableBuffering();
+                var remoteIp = context.Connection.RemoteIpAddress;
+                var body = await new StreamReader(context.Request.Body).ReadToEndAsync();
+                context.Request.Body.Seek(0, SeekOrigin.Begin);
+                if (_options.Value.EnableManaging)
+                {
+                    if (body.Contains("agent(address:\\\"") || body.Contains("agent(address: \\\""))
+                    {
+                        try
+                        {
+                            var agent = new Address(body.Split("\\\"")[1].Split("0x")[1]);
+                            UpdateIpSignerList(remoteIp!.ToString(), agent);
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.Error(
+                                "[GRAPHQL-MULTI-ACCOUNT-MANAGER] Error message: {message} Stacktrace: {stackTrace}",
+                                ex.Message,
+                                ex.StackTrace);
+                        }
+                    }
```

Comment:
Not critical to merge but it seems better to parse with regex or real GraphQL parser 🤔

Reply:
That sounds better. Let me look into this 👍

Reply:
@moreal I've looked into this and I was able to parse the request by doing something like this:

```
public class GraphQlRequest
{
    public string? Query { get; set; }
    public Dictionary<string, object>? Variables { get; set; }
}

GraphQlRequest graphQLRequest = JsonConvert.DeserializeObject<GraphQlRequest>(body);
```

However, the problem is, `JsonConvert.DeserializeObject()` takes alittle long to deserialize and I'm worried that it'll affect the overall response performance 😢.

I'll merge this for now and create an new issue to address a more efficient way to parse the data. 🙏

Reply:
Okay! ✅